.gitignore

```diff
@@ -0,0 +1,6 @@
+Makefile
+*.o
+*.so
+*.so.*
+*.moc
+moc_*
```

Comment:
Can you also include:
```
*~
*.[oa]
# Patches and their leftovers
*.orig
*.diff
*.rej
*.patch

*.moc.cpp

# Build folder and lsp server files
/build*/
/compile_commands.json
.clangd
.ccls-cache
.cache



# Qt Creator
*.user
```

Reply:
Let's not overdo this. We don't commonly play with patches here and if we did it might be still good to have those visible. We don't build with clang and for some I'm not sure where these are coming from. With the existing set I've gotten already rid of most of the unnecessary noise I'm getting with building locally.

Reply:
Clangd isn't the clang compiler but clang language server which yes we use.